subsystem/apiclient/kubeapi.go

```diff
@@ -0,0 +1,265 @@
+	}
+
+	metaName := uuid.New().String()
+	c := adiiov1alpha1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Cluster",
+			APIVersion: APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      metaName,
+			Namespace: kc.namespace,
+		},
+		Spec: adiiov1alpha1.ClusterSpec{
+			Name:                     clusterName,
+			OpenshiftVersion:         swag.StringValue(cParams.OpenshiftVersion),
+			BaseDNSDomain:            swag.StringValue(&cParams.BaseDNSDomain),
+			ClusterNetworkCidr:       swag.StringValue(cParams.ClusterNetworkCidr),
+			ClusterNetworkHostPrefix: swag.Int64Value(&cParams.ClusterNetworkHostPrefix),
+			ServiceNetworkCidr:       swag.StringValue(cParams.ServiceNetworkCidr),
+			IngressVip:               swag.StringValue(&cParams.IngressVip),
+			SSHPublicKey:             swag.StringValue(&cParams.SSHPublicKey),
+			VIPDhcpAllocation:        swag.BoolValue(cParams.VipDhcpAllocation),
+			HTTPProxy:                swag.StringValue(cParams.HTTPProxy),
+			HTTPSProxy:               swag.StringValue(cParams.HTTPSProxy),
+			NoProxy:                  swag.StringValue(cParams.NoProxy),
+			UserManagedNetworking:    swag.BoolValue(cParams.UserManagedNetworking),
+			AdditionalNtpSource:      swag.StringValue(cParams.AdditionalNtpSource),
+			PullSecretRef:            pullSecretRef,
+		},
+	}
+	if deployErr := kc.client.Create(ctx, &c); deployErr != nil {
+		return nil, errors.Wrapf(deployErr, "failed to deploy cluster")
+	}
+
+	key := types.NamespacedName{
+		Name:      metaName,
+		Namespace: kc.namespace,
+	}
+	cluster, getClusterErr := kc.getClusterByKey(key)
+	for i := 0; i < 5; i++ {
+		cluster, getClusterErr = kc.getClusterByKey(key)
+		if getClusterErr == gorm.ErrRecordNotFound {
+			time.Sleep(time.Second)
+		}
+	}
+	if getClusterErr != nil {
+		return nil, errors.Wrapf(getClusterErr, "failed to get cluster after creation")
+	}
+	return cluster, nil
+}
+
+func (kc *KubeAPIClient) deployPullSecret(
+	ctx context.Context,
+	name, pullSecret string,
+) (*corev1.SecretReference, error) {
+
+	secretName := fmt.Sprintf("%s-secret", name)
+	secretRef := &corev1.SecretReference{
+		Name:      secretName,
+		Namespace: kc.namespace,
+	}
+
+	key := types.NamespacedName{
+		Name:      secretName,
+		Namespace: kc.namespace,
+	}
+	secret := &corev1.Secret{}
+	if getSecretErr := kc.client.Get(ctx, key, secret); runtimeclient.IgnoreNotFound(getSecretErr) != nil {
+		return nil, getSecretErr
+	} else if getSecretErr == nil {
+		return secretRef, nil
+	}
+
+	if pullSecret == "" {
+		return nil, nil
+	}
+
+	data := map[string]string{"pullSecret": pullSecret}
+	secretData := make(map[string][]byte)
+	for k, v := range data {
+		secretData[k] = []byte(base64.StdEncoding.EncodeToString([]byte(v)))
+	}
+
+	secret = &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: kc.namespace,
+		},
+		Data: secretData,
```

Comment:
I think that you can use `StringData` and then you will not need to encrypt.
it doesn't work in unit-tests, but i did worked for me in minikube.